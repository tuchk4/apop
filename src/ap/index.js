import crop from './actions/splice';
import filter from './actions/filter';
import join from './actions/join';
import map from './actions/map';
import remove from './actions/remove';
import sort from './actions/sort';
import swap from './actions/swap';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };

const ap = Object.assign(apply, sourceActions);
const recursive = Object.assign(recursiveApply, sourceActions);

export default Object.assign(ap, { recursive });
