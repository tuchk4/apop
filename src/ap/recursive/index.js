import crop from '../actions/splice';
import filter from '../actions/filter';
import join from '../actions/join';
import map from '../actions/map';
import remove from '../actions/remove';
import sort from '../actions/sort';
import swap from '../actions/swap';

import { recursiveApply } from '../utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };
const recursiveActions = {};

Object.keys(sourceActions).forEach(action => {
  recursiveActions[action] = (...args) =>
    recursiveApply(sourceActions[action](...args));
});

const recursive = Object.assign(
  (...actions) => recursiveApply(...actions),
  recursiveActions
);

export default recursive;
