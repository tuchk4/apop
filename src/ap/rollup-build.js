// import { rollup } from 'rollup';
//
// import resolve from 'rollup-plugin-node-resolve';
// import uglify from 'rollup-plugin-uglify';
// import commonjs from 'rollup-plugin-commonjs';
// import babel from 'rollup-plugin-babel';
// import { minify } from 'uglify-es';
// import babelrc from 'babelrc-rollup';
//
// const babelConfig = {
//   presets: [['es2015', { loose: true, modules: false }, 'stage-0']],
//   plugins: ['external-helpers', 'transform-object-rest-spread'],
// };
//
// var rollUpConfig = {
//   format: 'umd',
//   sourceMap: 'inline',
//   plugins: [
//     resolve({
//       jsnext: true,
//       main: true,
//       browser: true,
//     }),
//     commonjs(),
//     babel(
//       babelrc({
//         addExternalHelpersPlugin: false,
//         config: babelConfig,
//         exclude: 'node_modules/**',
//       })
//     ),
//   ],
// };
//
// async function make(config) {
//   let bundle = await rollup({
//     entry: `./src/${config.moduleName}.js`,
//     plugins: config.plugins,
//   });
//
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.js`,
//     sourceMap: config.sourceMap,
//   });
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.min.js`,
//     sourceMap: config.sourceMap,
//     plugins: [...rollUpConfig.plugins, uglify({}, minify)],
//   });
//   console.log(`make ./${config.moduleName}.js and ./${config.moduleName}.min.js`);
// }
//
// (async function() {
//
//   const builds = ['rmkArray','recursive/rmkArray'];
//
//   builds.forEach(async moduleName => await make({...rollUpConfig, moduleName}));
//
// })();
