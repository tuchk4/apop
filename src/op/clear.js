import action from './actions/clear';
import { apply } from './utils/apply';

export default config => apply(action(config));
