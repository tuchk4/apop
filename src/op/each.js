import action from './actions/each';
import { apply } from './utils/apply';

export default config => apply(action(config));
