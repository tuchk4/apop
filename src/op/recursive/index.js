import actions from '../actions';

import { recursiveApply } from '../utils/apply';

const recursive = Object.assign((...args) => recursiveApply(...args), actions);

export default recursive;
