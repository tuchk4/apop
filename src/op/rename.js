import action from './actions/rename';
import { apply } from './utils/apply';

export default config => apply(action(config));
