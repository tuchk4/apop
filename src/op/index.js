import clear from './actions/clear';
import toSnakeCase from './actions/toSnakeCase';
import toCamelCase from './actions/toCamelCase';
import update from './actions/update';
import rename from './actions/rename';
import each from './actions/each';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { clear, toSnakeCase, toCamelCase, update, rename, each };
const hackActions = {};

Object.keys(sourceActions).forEach(key => {
  hackActions[key] = (...args) =>
    Object.assign(sourceActions[key](...args), { shortcut: true, key });
});

const op = Object.assign((...args) => apply(...args), hackActions);
const recursive = Object.assign(
  (...args) => recursiveApply(...args),
  hackActions
);

export default Object.assign(op, { recursive });
