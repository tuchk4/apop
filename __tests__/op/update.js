import ap from '../../src/build/ap';
import op from '../../src/build/op';

const before = Object.seal({
  date: new Date(Date.UTC(2001, 11, 3)),
});

const misc = {
  year: 2001,
  month: 11,
  day: 3,
};

it('Field function object', () => {
  const formula = op(
    op.update({
      year: localState => localState.date.getFullYear(),
      month: localState => localState.date.getMonth(),
      day: localState => localState.date.getDate(),
    })
  );

  const result = formula(before);

  expect(result).toMatchSnapshot();
});

test('Field static object', () => {
  const formula = op(
    op.update({
      year: misc.year,
      month: misc.month,
      day: misc.day,
    })
  );

  const result = formula(before);

  expect(result).toMatchSnapshot();
});

test('Field function array', () => {
  const formula = ap(
    op.update({
      year: localState => localState.date.getFullYear(),
      month: localState => localState.date.getMonth(),
      day: localState => localState.date.getDate(),
    })
  );

  const result = formula([before]);

  expect(result).toMatchSnapshot();
});

test('Field static array', () => {
  const additionalFields = {
    year: 2000,
    month: 1,
    day: 1,
  };

  const formula = ap(op.update(additionalFields));
  const result = formula([before]);

  expect(result).toMatchSnapshot();
});

// --

test('Update shortcut: Field function object', () => {
  const result = op.update({
    year: localState => localState.date.getFullYear(),
    month: localState => localState.date.getMonth(),
    day: localState => localState.date.getDate(),
  })(before);

  expect(result).toMatchSnapshot();
});

test('Update shortcut: Field static object', () => {
  const result = op.update({
    year: misc.year,
    month: misc.month,
    day: misc.day,
  })(before);

  expect(result).toMatchSnapshot();
});

test('Update shortcut: Field function array', () => {
  const result = ap(
    op.update({
      year: localState => localState.date.getFullYear(),
      month: localState => localState.date.getMonth(),
      day: localState => localState.date.getDate(),
    })
  )([before]);

  expect(result).toMatchSnapshot();
});

test('Update shortcut: Field static array', () => {
  const additionalFields = {
    year: 2000,
    month: 1,
    day: 1,
  };
  const result = ap(op.update(additionalFields))([before]);

  expect(result).toMatchSnapshot();
});
