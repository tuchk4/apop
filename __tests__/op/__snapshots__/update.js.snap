// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field function array 1`] = `
Array [
  Object {
    "date": 2001-12-03T00:00:00.000Z,
    "day": 3,
    "month": 11,
    "year": 2001,
  },
]
`;

exports[`Field function object 1`] = `
Object {
  "date": 2001-12-03T00:00:00.000Z,
  "day": 3,
  "month": 11,
  "year": 2001,
}
`;

exports[`Field static array 1`] = `
Array [
  Object {
    "date": 2001-12-03T00:00:00.000Z,
    "day": 1,
    "month": 1,
    "year": 2000,
  },
]
`;

exports[`Field static object 1`] = `
Object {
  "date": 2001-12-03T00:00:00.000Z,
  "day": 3,
  "month": 11,
  "year": 2001,
}
`;

exports[`Update shortcut: Field function array 1`] = `
Array [
  Object {
    "date": 2001-12-03T00:00:00.000Z,
    "day": 3,
    "month": 11,
    "year": 2001,
  },
]
`;

exports[`Update shortcut: Field function object 1`] = `
Object {
  "date": 2001-12-03T00:00:00.000Z,
  "day": 3,
  "month": 11,
  "year": 2001,
}
`;

exports[`Update shortcut: Field static array 1`] = `
Array [
  Object {
    "date": 2001-12-03T00:00:00.000Z,
    "day": 1,
    "month": 1,
    "year": 2000,
  },
]
`;

exports[`Update shortcut: Field static object 1`] = `
Object {
  "date": 2001-12-03T00:00:00.000Z,
  "day": 3,
  "month": 11,
  "year": 2001,
}
`;
